### PR TITLE
Add new flow shipping recommendation task

### LINKED ISSUE #33367: Implement the new flow to install and connect WCS in Setting page
**User story:**
As a new merchant
I want to focus on the most important setup tasks
So I can review some details later

**Description:**
With Jetpack installed and connected, alongside WCS, we have the opportunity to set things up on the users' behalf, and ask them to review and confirm, rather than start from scratch and think.

**Design overview (with new spotlight component):**
cJQeyyhx9PraQrY3YAtCGM-fi-705%3A80783

**Acceptance Criteria:**

- [ ] Implement the new flow to install and connect WCS.

159506263-a846740e-b536-4c7f-896f-398d4c2aa238

ref: https://github.com/woocommerce/woocommerce/issues/32318

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes #33367

This PR:

1. Changes `Plugins` component in `@woocommerce/components` to have optional skip button since the step 2 is mandatory. https://github.com/woocommerce/woocommerce/commit/8200daf8a83809893fa5cd6ed2e53e55c445a41b
2. Adds link from payment settings page https://github.com/woocommerce/woocommerce/commit/5ee6243f0b9942236047744a8cdf753580f3b294
3. Adds a "hidden" tasklist, `shipping-recommendation` task, and its related fill in https://github.com/woocommerce/woocommerce/commit/0b7b4c6ed5316033f78c367321de958a1637f6a7

### How to test the changes in this Pull Request:

1. Enable the `shipping-smart-defaults` feature via WCA Test helper.
2. Set up a store in US
3. Go to WooCommerce > Settings > Shipping
4. Wait for "Recommended shipping solutions" to appear and click on "Get started" or "Activate" on WooCommerce Shipping
5. Observe it gets redirected to step 2 of `admin.php?page=wc-admin&task=shipping-recommendation`task
6. Click on "Install & enable" and observe `woocommerce-services` and `jetpack` plugins are installed
7. Click on "Connect" on the next step and connect site to WPCOM
8. When redirected back, observe the task automatically redirects to `admin.php?page=wc-settings&tab=shipping&section=woocommerce-services-settings`

image

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33620/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Merge branch 'add/33367-new-flow-wcs-shipping-task' of github.com:woo…` f4c01604d6824edc83f61b7bad8554f3cf4be37b
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	 *
+	 * @return bool
+	 */
+	public static function has_plugins_active() {
+		return PluginsHelper::is_plugin_active( 'woocommerce-services' ) &&
+		PluginsHelper::is_plugin_active( 'jetpack' );
+	}
+
+	/**
+	 * Check if the store has physical products.
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php]: a small thing:

```suggestion
	 * Check if the Jetpack is connected.
```

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php [hunk 2]
```diff
@@ -282,6 +282,20 @@ public static function init_default_lists() {
 					self::get_list( 'extended_two_column' )
 				)
 			);
+
+			// Tasklist that will never be shown in homescreen,
+			// used for having tasks that are accessed by other means.
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php:287]: Nice approach!

File: packages/js/components/src/plugins/index.js [hunk 3]
```diff
@@ -18,7 +18,6 @@ export class Plugins extends Component {
 		};
 
 		this.installAndActivate = this.installAndActivate.bind( this );
-		this.skipInstaller = this.skipInstaller.bind( this );
```
Comment (chihsuan) [inline packages/js/components/src/plugins/index.js]: Can we still get the correct `this.` in `skipInstaller`?

File: plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/store-location.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+	if ( isResolving ) {
+		return null;
+	}
+
+	return (
+		<StoreLocationForm
+			onComplete={ ( values: { [ key: string ]: string } ) => {
+				const country = getCountryCode( values.countryState );
+				recordEvent( 'tasklist_tax_set_location', {
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/store-location.tsx]: Do we still need to emit this **tax** event in the shipping task?

File: packages/js/components/src/plugins/index.js (hunk 3 above)
Comment (ilyasfoo) [reply packages/js/components/src/plugins/index.js]: Whoops, this was accidental. Thanks for spotting!

File: plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/store-location.tsx (hunk 4 above)
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/store-location.tsx]: Nope, thanks for catching it! I think keeping the track but renaming `tax` to `shipping_recommendation` would be good for now.

File: plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/plugins.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+				'woocommerce'
+		  );
+
+	if ( isResolving ) {
+		return null;
+	}
+
+	return (
+		<>
+			<PluginInstaller
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/plugins.tsx:77]: I think this is replicated from the tax component, but I find it very confusing when we're not providing plugin slugs. 

I didn't know how it works until I checked the `PluginInstaller` code. 

https://github.com/woocommerce/woocommerce/blob/e9006b56dfc9747a23a5f8b0796b4a7e0576e14a/packages/js/components/src/plugins/index.js#L175-L180

Would you like to explicitly set the `pluginSlugs` prop?

**Review by chihsuan — DISMISSED** [review]
@ilyasfoo  Thanks for working on this! Looks good and tested well. 👍 

I think we could improve the plugin code readability by adding the `pluginSlugs` prop. 

Pre-approving!

**Comment by ilyasfoo** [comment]
Thanks for the review, @chihsuan! Excellent catches. I've made these commits to address your feedback:
- [Rename all tracks to shipping_recommendation](https://github.com/woocommerce/woocommerce/pull/33620/commits/a2e9bf460f841e9b5089d266df80667ef81bb51e)
- [Add back skip installer bind](https://github.com/woocommerce/woocommerce/pull/33620/commits/95b1bccf09b6daf40274f3bf0697d4c0f63b6f9c)
- [Pass plugins list to pluginSlugs to be more predictable](https://github.com/woocommerce/woocommerce/pull/33620/commits/62480123d9003199dede19374d2810bfa335f183)

File: plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/plugins.tsx (hunk 5 above)
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/tasks/fills/experimental-shipping-recommendation/components/plugins.tsx:77]: That's an excellent suggestion, I do think the implicit use of the default value is unreliable and less readable.

**Review by chihsuan — APPROVED** [review]
> I've made these commits to address your feedback:

Thanks for addressing it. Looks good! 💯

**Pull request merged**